src/Agent.Worker/Release/Artifacts/FileContainer/ContainerFetchEngine.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Microsoft.VisualStudio.Services.FileContainer;
+
+namespace Agent.Worker.Release.Artifacts.FileContainer
+{
+    public class ContainerFetchEngine : FetchEngine
+    {
+        public ContainerFetchEngine(
+            IExecutionContext logger,
```

Comment:
it's not only a logger, it also has the whole context on it, so still call it context or executioncontext.